File: binary_search/BinarySearch.java
package binary_search;
public class BinarySearch {
    public static void main(String[] args) {
        int[] nums = {1, 3, 5, 6, 7, 67};
        System.out.println(binarySearch(nums, 5));
    }

    public static int binarySearch(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;
        int middle, attempt;

        while (low <= high) {
            middle = (low + high) >>> 1;
            attempt = nums[middle];

            if (attempt == target) {
                return middle;
            }
            if (attempt > target) {
                high = middle -1;
                continue;
            }
            low = middle + 1;
        }

        return -1;
    }
}

File: binary_search/binary_search.js
const binary_search = ((nums, target) => {
    let low = 0;
    let high = nums.length - 1;

    while (low <= high) {
        let middle = parseInt((low + high) / 2);
        low + high / 2;
        let attempt = nums[middle];

        if (attempt == target) {
            return middle;
        }
        if (attempt > target) {
            high = middle - 1;
            continue;
        }
        low = middle + 1;
    }
    return null;
})

console.log(binary_search([1,3,4,5,67], 5));
